pkg/command/reconciler.go

```diff
@@ -445,12 +488,28 @@ func (r *Reconciliation) reconcileRunAlwaysCommand() (types.CommandStatus, error
 		return r.createChildJob()
 	}
 	if r.isStatusSet && r.isChildJobCompleted {
+		// Since this is for run always we need to delete
+		// and create child job
 		klog.V(1).Infof(
 			"Will delete command job: Command %q / %q",
 			r.command.GetNamespace(),
 			r.command.GetName(),
 		)
-		return r.deleteChildJob()
+		_, err := r.deleteChildJob()
+		if err != nil {
+			return types.CommandStatus{}, err
+		}
+		klog.V(1).Infof("Deleted command job: Command %q / %q successfully",
+			r.command.GetNamespace(),
+			r.command.GetName(),
+		)
+
+		klog.V(1).Infof(
```

Comment:
Can we mention as code comments etc that we dont want this to be eventual but immediate. We can explain that delete & create will happen in the same reconciliation process.

Reply:
Have we double checked if there will be no conflict during delete & immediate create.  The job name might conflict?
We might want to generate the job name or we might want to retry job create in some loop.

Reply:
Hmm, good catch!! Instead, I will wait till the object successfully deleted from etcd

Reply:
Updated PTAL